pandas/core/common.py

```diff
@@ -235,7 +236,16 @@ def asarray_tuplesafe(values: Iterable, dtype: NpDtype | None = None) -> ArrayLi
     if isinstance(values, list) and dtype in [np.object_, object]:
         return construct_1d_object_array_from_listlike(values)
 
-    result = np.asarray(values, dtype=dtype)
+    try:
+        with warnings.catch_warnings():
+            # Can remove warning filter once NumPy 1.24 is min version
+            warnings.simplefilter("ignore", np.VisibleDeprecationWarning)
+            result = np.asarray(values, dtype=dtype)
+    except ValueError:
```

Comment:
In general I think it's better to just wrap the line that can raise in the try block. Is it a problem with the warning catching in this case? No big deal, I guess the warnings stuff won't raise a `ValueError`, but if it does, the behavior won't be as expected.

Reply:
Hm in this case I think this is better as is, if we would wrap try-except into the catch_warnings statement, then we would also catch warnings I the except block, which isn't what we want here

Reply:
Yeah `result = np.asarray(values, dtype=dtype)` will raise a warning (due to our usage) w/ numpy < 1.24 and raise an exception with numpy >= 1.24. As mentioned, I don't want to accidentally mask a warning within the `except` block.